server/serverAPI/controller/ShoppingListController.ts

```diff
@@ -8,20 +8,53 @@ import ShoppingIngredientSchema from "../model/ingredient/requestSchema/Shopping
 
 import UnitSchema from "../model/unit/UnitSchema";
 import IUser from "../model/user/IUser";
-import BaseUserController from "./BaseUserController";
+import BaseIngredientController from "./BaseIngredientController";
 
 /**
  * This class creates several properties responsible for shopping list actions 
  * provided to the user.
  */
-export default class ShoppingListController extends BaseUserController {
+export default class ShoppingListController extends BaseIngredientController {
     private foodAPI: IIngredientAPI;
 
     constructor(database: IDatabase<IUser>, foodAPI: IIngredientAPI) {
         super(database);
         this.foodAPI = foodAPI;
     }
 
+    protected sortByRecipe(collection: IShoppingIngredient[], isReverse: boolean): any {
+        let recipeMap = new Map<string, IShoppingIngredient[]>();
+
+        let itemsWithoutRecipeID: IShoppingIngredient[] = [];
```

Comment:
I think these should be sorted lexicographically (forward or reverse specified by `isReverse`